sdk/storage/Azure.Storage.Blobs/src/AppendBlobClient.cs

```diff
@@ -330,6 +330,175 @@ protected sealed override BlobBaseClient WithCustomerProvidedKeyImpl(CustomerPro
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        public virtual async Task<Response<BlobContentInfo>> CreateIfNotExistsAsync(
+            BlobHttpHeaders? httpHeaders = default,
+            Metadata metadata = default,
+            AppendBlobAccessConditions? accessConditions = default,
+            CancellationToken cancellationToken = default) =>
+            await CreateIfNotExistsInternal(
+                httpHeaders,
+                metadata,
+                accessConditions ?? default,
+                true, // async
+                cancellationToken)
+                .ConfigureAwait(false);
+
+        /// <summary>
+        /// The <see cref="CreateIfNotExistsInternal"/> operation creates a new 0-length
+        /// append blob.  If the append blob already exists, the content of
+        /// the existing append blob will remain unchanged.  To add content to the append
+        /// blob, call the <see cref="AppendBlockAsync"/> operation.
+        ///
+        /// For more information, see <see href="https://docs.microsoft.com/rest/api/storageservices/put-blob" />.
+        /// </summary>
+        /// <param name="httpHeaders">
+        /// Optional standard HTTP header properties that can be set for the
+        /// new append blob.
+        /// </param>
+        /// <param name="metadata">
+        /// Optional custom metadata to set for this append blob.
+        /// </param>
+        /// <param name="accessConditions">
+        /// Optional <see cref="AppendBlobAccessConditions"/> to add
+        /// conditions on the creation of this new append blob.
+        /// </param>
+        /// <param name="async">
+        /// Whether to invoke the operation asynchronously.
+        /// </param>
+        /// <param name="cancellationToken">
+        /// Optional <see cref="CancellationToken"/> to propagate
+        /// notifications that the operation should be cancelled.
+        /// </param>
+        /// <returns>
+        /// A <see cref="Response{BlobContentInfo}"/> describing the
+        /// newly created append blob.
+        /// </returns>
+        /// <remarks>
+        /// A <see cref="StorageRequestFailedException"/> will be thrown if
+        /// a failure occurs.
+        /// </remarks>
+        private async Task<Response<BlobContentInfo>> CreateIfNotExistsInternal(
+            BlobHttpHeaders? httpHeaders,
+            Metadata metadata,
+            AppendBlobAccessConditions accessConditions,
+            bool async,
+            CancellationToken cancellationToken)
+        {
+            if (accessConditions.HttpAccessConditions.HasValue)
+            {
+                var ModifiedHttpAccessConditions = (HttpAccessConditions)accessConditions.HttpAccessConditions.Value;
+                ModifiedHttpAccessConditions.IfNoneMatch = new ETag(Constants.Wildcard);
```

Comment:
I think this ETag tells the server to send back a 304 if the server has the resource with any ETag present. Looking at the recorded tests, it seems the server always returns a 409 and we throw the StorageRequestFailedException with the AlreadyExists code. Do you know when we would expect it to return 304/can we add a test for this?

Reply:
https://docs.microsoft.com/en-us/rest/api/storageservices/specifying-conditional-headers-for-blob-service-operations#Subheading3 has the details.

Reply:
Thanks for the pointer. This says it would return a 412 if the preconditions aren't met, which makes me wonder why the responses from the tests always return a 409.  In any case, would we also have to handle a different error code if the server really does return a 412?

Reply:
Yeah I will add something in to catch the 412 error. I am also looking into sending like a Get operation (e.g. GetProperties) and that should kick us back a 304 code. But that would require sending two requests to the service instead of one, if the blob does not exist yet.

Reply:
But would we need to handle this error if we remove the ability to set the HttpAccessConditions as per a different comment?

Reply:
Sending two requests would rustle Jeff's jimmies more than I care to _(plus then you've got to worry about race conditions between getting properties and the subsequent create, ...)_.  I wouldn't even worry about the 412 if the ContainerAlreadyExists error code is being set on whatever's returned.